### ISSUE TITLE
Employing LeakyReLU as the activation function of my CNN model causes 'nan' loss during training?

### ISSUE THREAD

**Issue by talhakabakus** [keras-team/keras#14576]
When I change my `CNN` model's activation function, which is `ReLU`, to `LeakyReLU`, both training and validation losses become `nan`. How can I resolve this issue?

Here is my model's summary:
```
Shape of all data: (1889, 10801)
Shape of X_train: (1322, 10800, 1)
Shape of Y_train: (1322, 3)
Shape of X_test: (567, 10800, 1)
Shape of y_test: (567, 3)
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
conv1d_48 (Conv1D)           (None, 10721, 128)        10368     
_________________________________________________________________
batch_normalization_48 (Batc (None, 10721, 128)        512       
_________________________________________________________________
activation_48 (Activation)   (None, 10721, 128)        0         
_________________________________________________________________
max_pooling1d_41 (MaxPooling (None, 5360, 128)         0         
_________________________________________________________________
dropout_29 (Dropout)         (None, 5360, 128)         0         
_________________________________________________________________
conv1d_49 (Conv1D)           (None, 5357, 128)         65664     
_________________________________________________________________
batch_normalization_49 (Batc (None, 5357, 128)         512       
_________________________________________________________________
activation_49 (Activation)   (None, 5357, 128)         0         
_________________________________________________________________
max_pooling1d_42 (MaxPooling (None, 2678, 128)         0         
_________________________________________________________________
dropout_30 (Dropout)         (None, 2678, 128)         0         
_________________________________________________________________
conv1d_50 (Conv1D)           (None, 2675, 128)         65664     
_________________________________________________________________
batch_normalization_50 (Batc (None, 2675, 128)         512       
_________________________________________________________________
activation_50 (Activation)   (None, 2675, 128)         0         
_________________________________________________________________
max_pooling1d_43 (MaxPooling (None, 1337, 128)         0         
_________________________________________________________________
dropout_31 (Dropout)         (None, 1337, 128)         0         
_________________________________________________________________
conv1d_51 (Conv1D)           (None, 1334, 256)         131328    
_________________________________________________________________
batch_normalization_51 (Batc (None, 1334, 256)         1024      
_________________________________________________________________
activation_51 (Activation)   (None, 1334, 256)         0         
_________________________________________________________________
max_pooling1d_44 (MaxPooling (None, 667, 256)          0         
_________________________________________________________________
global_max_pooling1d_6 (Glob (None, 256)               0         
_________________________________________________________________
dense_20 (Dense)             (None, 512)               131584    
_________________________________________________________________
dense_21 (Dense)             (None, 3)                 1539      
=================================================================
```

And, the model was compiled as follows:
```
n_lr = 1e-5
weight_decay = 1e-4
adam = Adam(lr=n_lr, beta_1=0.9, beta_2=0.999, epsilon=1e-08, decay=weight_decay)
model.compile(loss='categorical_crossentropy', optimizer=adam, metrics=['acc'])
```
p.s. The model is deployed on _Google Colab_.

Feel free to ask further questions regarding the model.

**Comment by bguetarni** [comment]
This can be due to the exploding gradient problem. Since you allowed your neurons activation to be negative with LeakyReLU, the model could pushed them to `-inf`. To overcome exploding gradient, try to use the `clip_norm` (see **kwargs arguments of Adam) for gradient clipping.

**Comment by jvishnuvardhan** [comment]
@talhakabakus Sorry for the late response. Is this still an issue for you?

Can you follow the above suggestion and let us know whether the issue resolved. If it was not resolved even with @bguetarni suggestion, please share a simple standalone code to reproduce the issue. Thanks!
